src/plugins/playback-speed/renderer.tsx

```diff
@@ -0,0 +1,116 @@
+import { render } from 'solid-js/web';
+
+import { createSignal } from 'solid-js';
+
+import { getSongMenu } from '@/providers/dom-elements';
+
+import { PlaybackSpeedSlider } from './components/slider';
+import { t } from '@/i18n';
+
+import { isMusicOrVideoTrack } from '@/plugins/utils/renderer/check';
+
+const MIN_PLAYBACK_SPEED = 0.07;
+const MAX_PLAYBACK_SPEED = 16;
+
+const forcePlaybackRate = (e: Event) => {
+  if (e.target instanceof HTMLVideoElement) {
+    const videoElement = e.target;
+    if (videoElement.playbackRate !== speed()) {
+      videoElement.playbackRate = speed();
+    }
+  }
+};
+
+const roundToTwo = (n: number) => n.toFixed(2);
+
+const [speed, setSpeed] = createSignal(1);
+const sliderContainer = document.createElement('div');
+
+export const onPlayerApiReady = () => {
+  const observePopupContainer = () => {
+    const updatePlayBackSpeed = () => {
+      const videoElement = document.querySelector<HTMLVideoElement>('video');
+      if (videoElement) {
+        videoElement.playbackRate = speed();
+      }
+
+      setSpeed(speed());
+    };
+
+    render(
+      () => (
+        <PlaybackSpeedSlider
+          speed={speed()}
+          title={t('plugins.playback-speed.templates.button')}
+          onImmediateValueChanged={(e) => {
+            let targetSpeed = Number(e.detail.value ?? MIN_PLAYBACK_SPEED);
+            
+            if (isNaN(targetSpeed)) {
+              targetSpeed = 1;
+            }
+            
+            targetSpeed = Math.min(Math.max(MIN_PLAYBACK_SPEED, targetSpeed), MAX_PLAYBACK_SPEED);
+            
+            setSpeed(targetSpeed);
+            updatePlayBackSpeed();
+          }}
+          onWheel={(e) => {
+            e.preventDefault();
+
+            if (isNaN(speed())) {
+              setSpeed(1);
+            }
+
+            // E.deltaY < 0 means wheel-up
+            setSpeed((prev) =>
+              roundToTwo(
+                e.deltaY < 0
+                  ? Math.min(prev + 0.01, MAX_PLAYBACK_SPEED)
+                  : Math.max(prev - 0.01, MIN_PLAYBACK_SPEED),
+              ),
+            );
+
+            updatePlayBackSpeed();
+          }}
+        />
+      ),
+      sliderContainer,
+    );
+
+    const observer = new MutationObserver(() => {
+      const menu = getSongMenu();
+
+      if (menu && !menu.contains(sliderContainer) && isMusicOrVideoTrack()) {
+        menu.prepend(sliderContainer);
+      }
+    });
+
+    const popupContainer = document.querySelector('ytmusic-popup-container');
+    if (popupContainer) {
+      observer.observe(popupContainer, {
+        childList: true,
+        subtree: true,
+      });
+    }
```

Comment:
the observer need to be disconnected when the plugin is unloaded

How about this?
```ts
const unloads: (() => void)[] = [];

// ...
const onPlayerApiReady = () => {
  // ...
  
	if (popupContainer) {
	  observer.observe(popupContainer, {
	    childList: true,
			subtree: true,
	  });
		unloads.push(observer.disconnect);
	}

  // ...
};

export const onUnload = () => {
	unloads.forEach((unload) => unload());
  // ...
};
```